File: models/variableincome.js
'use strict';
const { Model } = require('sequelize');
module.exports = (sequelize, DataTypes) => {
	class VariableIncome extends Model {
		static associate(models) {
			VariableIncome.hasOne(models.Client, { foreignKey: 'customerId' });
		}
	}
	VariableIncome.init(
		{
			customerId: {
				allowNull: false,
				autoIncrement: false,
				primaryKey: true,
				type: DataTypes.STRING,
			},
			bankId: DataTypes.INTEGER,
			name: DataTypes.STRING,
			type: DataTypes.STRING,
			exempt: DataTypes.BOOLEAN,
			minimalValue: DataTypes.INTEGER,
			maturity: DataTypes.STRING,
			date: DataTypes.DATEONLY,
		},
		{
			sequelize,
			modelName: 'VariableIncome',
		}
	);
	return VariableIncome;
};

File: src/models/variableincome.py
from pydantic import BaseModel
from datetime import date

class VariableIncome(BaseModel):
    
    customer_id: int
    bank_id: int
    name: str
    type: str
    exempt: bool
    interest_rate: float
    maturity: date
    date: date
    price: float
    rate: float